src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java

```diff
@@ -667,4 +675,49 @@ private String getFileName(File f) {
             return fileName;
         }
     }
+
+    /**
+     * issues #1111
+     * Check the recipe of deleted component, if it uses docker image and the image not used by other, delete the image.
+     *
+     * @param componentIdentifier  identifier for the component
+     * @throws PackageLoadingException this may from findComponentRecipeContent
+     */
+    private void removeDockerImage(ComponentIdentifier componentIdentifier) throws PackageLoadingException {
+        logger.atDebug("removeDockerImage-start").log();
+        List<String> allRecipeContentList = componentStore.getAllComponentRecipeContent();
+        Optional<String> componentRecipeContent = componentStore.findComponentRecipeContent(componentIdentifier);
+        List<String> toBeRemovedList = new ArrayList<String>();
+        if (componentRecipeContent.isPresent()) {
+            RecipeLoader.RecipeFormat format = RecipeLoader.RecipeFormat.YAML;
+            // the recipe is JSON or YML
+            if (componentRecipeContent.get().startsWith("{")) {
+                format = RecipeLoader.RecipeFormat.JSON;
+            }
+            // get recipe
+            com.amazon.aws.iot.greengrass.component.common.ComponentRecipe componentRecipe =
+                    RecipeLoader.parseRecipe(componentRecipeContent.get(), format);
+            List<String> currentUriList =
+                    componentRecipe.getManifests().stream()
+                            .map(PlatformSpecificManifest::getArtifacts)
+                            .flatMap(Collection::stream)
+                            .map(item -> item.getUri().toString())
```

Comment:
You can filter directly here on the URI scheme, see [here](https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloaderFactory.java#L81)